internal/tpm/tpm.go

```diff
@@ -169,6 +179,93 @@ func (t *TPM) CreateLDevID(srk tpm2.NamedHandle) (*tpm2.NamedHandle, error) {
 	return t.ldevid, nil
 }
 
+func (t *TPM) GetLDevIDPubKey() (crypto.PublicKey, error) {
+	if t.ldevid == nil {
+		return nil, fmt.Errorf("ldevid not initialized")
+	}
+
+	pub, err := tpm2.ReadPublic{
+		ObjectHandle: t.ldevid.Handle,
+	}.Execute(transport.FromReadWriter(t.conn))
+	if err != nil {
+		return nil, fmt.Errorf("could not read public key: %v", err)
+	}
+	outpub, err := pub.OutPublic.Contents()
+	if err != nil {
+		return nil, fmt.Errorf("could not get contents of TPM2Bpublic: %v", err)
+	}
+	if outpub.Type != tpm2.TPMAlgECC {
+		return nil, fmt.Errorf("public key alg %d for ldevid key is unsupported", outpub.Type)
+	}
+	details, err := outpub.Parameters.ECCDetail()
+	if err != nil {
+		return nil, fmt.Errorf("cannot read ecc details for ldevid key: %v", err)
+	}
+	curve, err := details.CurveID.Curve()
+	if err != nil {
+		return nil, fmt.Errorf("could not get curve id for ldevid key: %v", err)
+	}
+	unique, err := outpub.Unique.ECC()
+	if err != nil {
+		return nil, fmt.Errorf("could not get unique parameters for ldevid key: %v", err)
+	}
+	pubkey := &ecdsa.PublicKey{
+		Curve: curve,
+		X:     new(big.Int).SetBytes(unique.X.Buffer),
+		Y:     new(big.Int).SetBytes(unique.Y.Buffer),
+	}
+	// converts ecdsa.PublicKey to crypto.PublicKey
+	t.ldevidPub = pubkey
+	return pubkey, nil
+}
+
+func (t TPM) Public() crypto.PublicKey {
+	return t.ldevidPub
```

Comment:
this seems strange why not jsut have a single way to GetLDevIDPubKey

Reply:
it ensures the key seems useful why not always use it and return the key if it exists?

Reply:
I think doing this: https://github.com/flightctl/flightctl/pull/1409/files/5ac82085a1ff805777f4cfcd13f8fc755e50f0d9#r2222993702 would simplify a lot (including this discussion) would you like me to do that now? Or as a separate PR? 

If we have `Public` first call `GetLDevIDKeyPublic` then we should probably implement some sort of method for not redoing all of the validation if there's no need for it. It also wouldn't be able to do anything with errors other than panic.

Reply:
followup is fine but we need it soon so we don't forget